File: Advanced Web Technology/Lab/Lab-11 08-10 Assignment-11/Task3.php
<!-- 3.	Create a PHP constructor to set employee name, designation, basic salary, allowances amount and deduction amount. Display gross salary and net salary using destructor. -->


<?php
class Employee {
    private $name;
    private $designation;
    private $basicSalary;
    private $allowances;
    private $deductions;

    public function __construct($name, $designation, $basicSalary, $allowances, $deductions) {
        $this->name = $name;
        $this->designation = $designation;
        $this->basicSalary = $basicSalary;
        $this->allowances = $allowances;
        $this->deductions = $deductions;
    }

    public function __destruct() {
        $grossSalary = $this->basicSalary + $this->allowances;
        $netSalary = $grossSalary - $this->deductions;
        echo "Employee Name: {$this->name}\n";
        echo "Designation: {$this->designation}\n";
        echo "Gross Salary: {$grossSalary}\n";
        echo "Net Salary: {$netSalary}\n";
    }
}

$emp = new Employee("Saichandan", "Software Engineer", 40000000000, 100000000, 50000);
?>

File: 2025-26/SEM4/Object Oriented Programming using C++/Lab/Solutions/Lab-3 28-01-26 Assignment 3/task3.cpp
#include <iostream>
using namespace std;

float gross_salary(float bs, float hra, float da, float ta)
{
    return bs + hra + da + ta;
}

void net_salary(float gross_sal, float pt, float pf)
{
    float net = gross_sal - (pt + pf);
    cout << "Gross Salary: " << gross_sal << endl;
    cout << "Net Salary: " << net << endl;
}

int main()
{
    float bs, hra, da, ta, pt, pf;

    cout << "Enter Basic Salary: ";
    cin >> bs;
    cout << "Enter HRA, DA, and TA: ";
    cin >> hra >> da >> ta;
    cout << "Enter PT and PF: ";
    cin >> pt >> pf;

    float gs = gross_salary(bs, hra, da, ta);
    net_salary(gs, pt, pf);

    return 0;
}